pump/storage/vlog.go

```diff
@@ -317,6 +317,7 @@ func (vlog *valueLog) write(reqs []*request) error {
 	vlog.filesLock.RUnlock()
 
 	var bufReqs []*request
+	vlog.buf.Reset()
```

Comment:
`Reset` when entering this function VS when leaving this function, which is better?

Reply:
It makes little difference in our set up. Since my intention is to never use any data left over in the buffer, I think it's clearer to reset the buffer upfront.